x-pack/elastic-agent/pkg/core/monitoring/config/config.go

```diff
@@ -39,6 +46,5 @@ func DefaultConfig() *MonitoringConfig {
 			Port:    defaultPort,
 		},
 		Namespace: defaultNamespace,
-		Pprof:     false,
```

Comment:
@ruflin, I changed this from a default false behaviour to default nil so that a value inserted into fleet.yml would be used when the `Reload` function is called (so the setting is now applied to beats that the agent starts)

Reply:
Interesting, so value only gets overwritten if not set yet?

@narph This likely should also feed into your thoughts on how these files should work in the future.